Security fix #4.  Zero the premaster secret after the secret key is
computed when doing Diffie-Hellman.

--- openssl-0.9.6/ssl/s3_srvr.c	Mon Sep 11 08:42:59 2000
+++ openssl-0.9.6a/ssl/s3_srvr.c	Thu Apr  5 16:14:59 2001
@@ -1414,6 +1414,7 @@
 		s->session->master_key_length=
 			s->method->ssl3_enc->generate_master_secret(s,
 				s->session->master_key,p,i);
+		memset(p,0,i);
 		}
 	else
 #endif
